### DISCUSSION TITLE
Database configuration

### DISCUSSION THREAD

**Discussion by ndiayepi** [spring-petclinic/spring-petclinic-microservices#235] [General]
Hello, I am experimenting with petclinic-microservices but there is a crutial point that I cannot understand. How to use its own configuration for a mysql database (IP, port ...)? I understand that you need to modify the config repository and I cloned this config repository first but I can't get the microservices to point to use this new config.
Thank you in advance for your help.
Best regards,
Pierre

**Comment by arey** [comment]
Hi Pierre. I suppose you have read the `Database configuration` section from the readme: https://github.com/spring-petclinic/spring-petclinic-microservices#database-configuration

By default, the `mysql` Spring profile uses the `jdbc:mysql://localhost:3306/petclinic?useSSL=false` JDBC connection URL:
https://github.com/spring-petclinic/spring-petclinic-microservices-config/blob/main/application.yml

To change the host and port of the database, you have to clone the `spring-petclinic-microservices-config` repository, change the JDBC connection URL then reference your repository with one of the both following options:

1. Either you can tell Config Server to use your local Git repository by using native Spring profile and setting GIT_REPO environment variable, for example: -Dspring.profiles.active=native -DGIT_REPO=/projects/spring-petclinic-microservices-config
2. Or you can change the Config Server URL from https://github.com/spring-petclinic/spring-petclinic-microservices/blob/master/spring-petclinic-config-server/src/main/resources/application.yml

I hope this help you. If so we could add some information into the readme.

**Reply by ndiayepi** [reply]
Hello Antoine,
Yes I have read the Database configuration section and despite the fact that I followed the instructions, I have the impression that the application uses the same configuration file.
In fact, here is what I did:
**1.** I created a local repository in which I copied the config repository files "https://github.com/spring-petclinic/spring-petclinic-microservices-config".
**2.** I modified the "application.yml" file in my local repository then I committed these modifications.
**3.** I modified the "application.yml" file located in the src/main/resources folder of the "spring-petclinic-config-server" project by putting the following content:
spring:
   clouds:
     config:
       server: server
         git:
           uri: https://github.com/ndiayepi/pn-spring-petclinic-microservices-config
           default-label: main
         # Use the File System Backend to avoid git pulling. Enable "native" profile in the Config Server.
         native:
           searchLocations: file:///${GIT_REPO}

**4.** I made a commit of the changes made.
**5.** First I started the configuration service by running the command "../mvnw spring-boot:run -Dspring.profiles.active=native -DGIT_REPO=/MY_HOME_USER_DIR/petclinic/localrepo
**6.** The then started the other services (discovery, vet-service, ...)
**7.** In the config server logs, I saw the message "Adding property source: Config resource 'file [/tmp/config-repo-1233029556764960973/application.yml]' via location 'file:/tmp/config- repo-1233029556764960973/' (document #0)"
This means that the config server stores the files in this temporary directory. Off by opening the "application.yml" file found in this temporary directory I see that it is different from the one found in my local repository; which means that the config server does not use my config files from the local repository.

So I'd like to know if I missed something that causes my local config repository to not be used when I've done everything required for it to be used.
Thank you again for your help.
Best regards,
Pierre

**Comment by arey** [comment]
Hi Pierre,

If you use the `spring-boot` maven plugin to start the Config Server, you could use the `spring-boot.run.jvmArguments` to enable the `native`profile and pass the `GIT_REPO` variable
```
mvn spring-boot:run -Dspring-boot.run.jvmArguments="-Dspring.profiles.active=native -DGIT_REPO=/MY_HOME_USER_DIR/petclinic/localrepo" 
```

Regards

**Comment by ndiayepi** [comment]
Hello Antoine,
Thanks a lot, it's working now! I am really grateful for your help!
Regards,
Pierre

**Comment by arey** [comment]
Glad I could help
See you soon :D
